optics-core/src/Optics/Traversal.hs

```diff
@@ -470,6 +480,81 @@ adjoin4 o1 o2 o3 o4 = combined % traversed
       []       ->            pure a
 {-# INLINE [1] adjoin4 #-}
 
+----------------------------------------
+
+-- | Traverse immediate occurrences of the type @a@ within the type @s@ using
```

Comment:
What does "immediate" mean here?

Reply:
That when it sees `a`, it won't try to traverse it further in search for another `a` within it (though I'm not sure if that's even possible). I'm not sure how to express that better :thinking:

Reply:
How about dropping "immediate", and adding a sentence along the lines of "If `a` occurs recursively in its own definition, only outermost occurrences of `a` within `s` will be traversed." For example:
```hs
>>> toListOf gplate "foo" :: [String]
["oo"]
>>> toListOf gplate ("foo","bar") :: [String]
["foo","bar"]
```

Reply:
Done.